src/goose/utils/safe_mode.py

```diff
@@ -0,0 +1,12 @@
+import os
+
+
+GOOSE_SAFE_MODE_ENV = "GOOSE_SAFE_MODE"
+
+def set_safe_mode() -> None:
+    os.environ[GOOSE_SAFE_MODE_ENV] = "true"
+    return None
+
+def is_in_safe_mode() -> bool:
+    return os.getenv(GOOSE_SAFE_MODE_ENV, "false") == "true"
```

Comment:
just an opinion i think adding "true" "false" clutters the environment. imo we should should only check if it exists or not (e.g. `unset GOOSE_SAFE_MODE` would disable _safe mode_). 

this minimizes user error